commit d05b773cc3f41c372f913649609db160d68e03ad
Author: Eric Anholt <eric@anholt.net>
Date:   Tue Aug 8 13:56:05 2017 -0700

    drm/vc4: Fix leak of HDMI EDID
    
    We don't keep a pointer to it around anywhere, so it's our job to free
    it.
    
    Cc: Stefan Wahren <stefan.wahren@i2se.com>
    Link: https://github.com/anholt/linux/issues/101
    Fixes: c8b75bca92cb ("drm/vc4: Add KMS support for Raspberry Pi.")
    Signed-off-by: Eric Anholt <eric@anholt.net>
    Link: https://patchwork.freedesktop.org/patch/msgid/20170808205605.4432-1-eric@anholt.net
    Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Tested-by: Stefan Wahren <stefan.wahren@i2se.com>
    (cherry picked from commit 5afe0e62a548a0aa7d2095a98530b68f3d2c64e6)

diff --git a/drivers/gpu/drm/vc4/vc4_hdmi.c b/drivers/gpu/drm/vc4/vc4_hdmi.c
index 50ba36ffde9c..48ded8cf45fc 100644
--- a/drivers/gpu/drm/vc4/vc4_hdmi.c
+++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
@@ -243,6 +243,7 @@ static int vc4_hdmi_connector_get_modes(struct drm_connector *connector)
 	drm_mode_connector_update_edid_property(connector, edid);
 	ret = drm_add_edid_modes(connector, edid);
 	drm_edid_to_eld(connector, edid);
+	kfree(edid);
 
 	return ret;
 }
